### PR TITLE
[ScreenTime] New framework for Xcode 12 up to beta 4.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/screentime.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	[Mac (11, 0), iOS (14, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface STScreenTimeConfiguration : NSSecureCoding {
+		[Export ("enforcesChildRestrictions")]
+		bool EnforcesChildRestrictions { get; }
+	}
+
+	[Mac (11,0), iOS (14,0)]
+	[BaseType (typeof(NSObject))]
```
Comment (mandel-macaque) [inline src/screentime.cs:24]:
```suggestion
	[BaseType (typeof (NSObject))]
```

File: src/screentime.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+		[Export ("stopObserving")]
+		void StopObserving ();
+
+		[NullAllowed, Export ("configuration", ArgumentSemantic.Strong)]
+		STScreenTimeConfiguration Configuration { get; }
+	}
+
+	[Mac (11, 0), iOS (14, 0)]
+	[BaseType (typeof (NSObject))]
```
Comment (spouliot) [inline src/screentime.cs:42]: `[DisableDefaultCtor]` ?
not clear (check headers) if a bindle identifier is required (or not)
In doubt, disable - easier to add later than remove :)

File: src/screentime.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+		[Export ("deleteHistoryDuringInterval:")]
+		void DeleteHistory (NSDateInterval interval);
+
+		[Export ("deleteAllHistory")]
+		void DeleteAllHistory ();
+	}
+
+	[Mac (11, 0), iOS (14, 0)]
+	[BaseType (typeof (UIViewController))]
```
Comment (spouliot) [inline src/screentime.cs:59]: same wrt `init`

File: src/screentime.cs (hunk 2 above)
Comment (mandel-macaque) [reply src/screentime.cs:42]:
```suggestion
	[BaseType (typeof (NSObject))]
	[DisableDefaultCtor]
```

File: src/screentime.cs (hunk 3 above)
Comment (mandel-macaque) [reply src/screentime.cs:59]:
```suggestion
	[BaseType (typeof (UIViewController))]
	[DisableDefaultCtor]
```

**Comment by mandel-macaque** [comment]
Failure is unrelated: https://github.com/xamarin/maccore/issues/1028

**Pull request merged**